pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/util/KitUtil.java

```diff
@@ -170,97 +170,110 @@ public class KitUtil {
+                                            kitLabel.getKitRequestSettings().getCollaboratorSampleTypeOverwrite();
                                 }
                                 if (collaboratorParticipantId == null) {
                                     logger.warn("CollaboratorParticipantId was too long " + ddpParticipant.getParticipantId());
                                 } else {
-                                    updateCollaboratorIds(kitLabelTriggered, collaboratorParticipantId, bspCollaboratorSampleType);
+                                    updateCollaboratorIds(kitLabel, collaboratorParticipantId, bspCollaboratorSampleType);
                                 }
                             }
                         } else {
-                            logger.error("Kit of pt  " + kitLabelTriggered.getDdpParticipantId() + " w/ kit id "
-                                    + kitLabelTriggered.getDsmKitId() + " is missing collaborator id");
+                            logger.error("Kit of pt  " + kitLabel.getDdpParticipantId() + " w/ kit id "
+                                    + kitLabel.getDsmKitId() + " is missing collaborator id");
                         }
                     }
                 }
             } catch (Exception e) {
-                throw new RuntimeException("Couldn't get address for participant " + kitLabelTriggered.getDdpParticipantId() + " in study "
-                        + kitLabelTriggered.getInstanceName() + " w/ kit id " + kitLabelTriggered.getDsmKitId(), e);
+                // note error and skip ptp
+                logger.error("Error getting address for participant {} in study {} with kit ID {}",
+                        kitLabel.getDdpParticipantId(), kitLabel.getInstanceName(), kitLabel.getDsmKitId(), e);
             }
             if (toAddress != null) {
-                buyShipmentForKit(easyPostUtil, kitLabelTriggered.getDsmKitId(), kitLabelTriggered.getKitRequestSettings(),
-                        kitLabelTriggered.getKitTyp(), toAddress.getId(), kitLabelTriggered.getBillingReference(), ddpInstanceDto);
+                //TODO: why not pass toAddress? - DC
+                buyShipmentForKit(easyPostUtil, kitLabel.getDsmKitId(), kitLabel.getKitRequestSettings(),
+                        kitLabel.getKitTyp(), toAddress.getId(), kitLabel.getBillingReference(), ddpInstanceDto);
             }
         }
 
         DBUtil.updateBookmark(0, BOOKMARK_LABEL_CREATION_RUNNING);
     }
 
+    protected static DDPParticipant getDDPParticipant(String ddpParticipantId, DDPInstance ddpInstance) {
+        Map<String, Map<String, Object>> participantESData = ElasticSearchUtil.getFilteredDDPParticipantsFromES(
+                ddpInstance, ElasticSearchUtil.BY_GUID + ddpParticipantId);
+        if (participantESData == null || participantESData.isEmpty()) {
+            participantESData = ElasticSearchUtil.getFilteredDDPParticipantsFromES(ddpInstance,
```

Comment:
Maybe a though for next refactoring
If we run  into this scenario frequently wondering if there is something in DSM DB to check and make this elastic call rather than doing other call to avoid additional elastic search call to improve performance.

Reply:
Good point. 
Thx for the suggestion.